src/dataframe/dataframe.jl

```diff
@@ -1016,25 +932,35 @@ function select!(df::DataFrame, inds::AbstractVector{Int})
     if !allunique(inds)
         throw(ArgumentError("indices must not contain duplicates"))
     end
+
     targetnames = _names(df)[inds]
-    deletecols!(df, setdiff(axes(df, 2), inds))
+    for i in setdiff(ncol(df):-1:1, inds)
+        splice!(_columns(df), i)
+        delete!(index(df), i)
+    end
     permutecols!(df, targetnames)
 end
 
 select!(df::DataFrame, c::Int) = select!(df, [c])
 select!(df::DataFrame, c::Any) = select!(df, index(df)[c])
 
 """
-    select(df::DataFrame, inds, copycols::Bool=true)
+    select(df::AbstractDataFrame, inds, copycols::Bool=true)
 
-Create a new `DataFrame` that contains columns from `df` specified by `inds` and return it.
+If `df` is a data frame create a new data frame that contains columns from `df`
```

Comment:
So `df` is not necessarily a data frame? :-/

Reply:
fixed